# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/sandbox/win/src/crosscall_server.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 2
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/sandbox/win/src/crosscall_server.h
--- a/security/sandbox/chromium/sandbox/win/src/crosscall_server.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/sandbox/win/src/crosscall_server.h	Wed Jun 12 11:10:48 2019 +0100
@@ -52,8 +52,8 @@
 // context: a user-defined pointer that was set using  ThreadProvider
 // reason: 0 if the callback was fired because of a timeout.
 //         1 if the callback was fired because of an event.
-typedef void (__stdcall * CrossCallIPCCallback)(void* context,
-                                                unsigned char reason);
+typedef void(__stdcall* CrossCallIPCCallback)(void* context,
+                                              unsigned char reason);
 
 // ThreadProvider models a thread factory. The idea is to decouple thread
 // creation and lifetime from the inner guts of the IPC. The contract is
@@ -77,7 +77,8 @@
   //           when the waitable object fires
   // context: a user-provider pointer that is passed back to the callback
   //          when its called
-  virtual bool RegisterWait(const void* client, HANDLE waitable_object,
+  virtual bool RegisterWait(const void* client,
+                            HANDLE waitable_object,
                             CrossCallIPCCallback callback,
                             void* context) = 0;
 
@@ -94,7 +95,7 @@
  public:
   // Factory constructor. Pass an IPCbuffer (and buffer size) that contains a
   // pending IPCcall. This constructor will:
-  // 1) validate the IPC buffer. returns NULL is the IPCbuffer is malformed.
+  // 1) validate the IPC buffer. returns nullptr is the IPCbuffer is malformed.
   // 2) make a copy of the IPCbuffer (parameter capture)
   static CrossCallParamsEx* CreateFromBuffer(void* buffer_base,
                                              uint32_t buffer_size,
@@ -102,7 +103,7 @@
 
   // Provides IPCinput parameter raw access:
   // index : the parameter to read; 0 is the first parameter
-  // returns NULL if the parameter is non-existent. If it exists it also
+  // returns nullptr if the parameter is non-existent. If it exists it also
   // returns the size in *size
   void* GetRawParameter(uint32_t index, uint32_t* size, ArgType* type);
 
@@ -169,7 +170,7 @@
 // Models an entity that can process an IPC message or it can route to another
 // one that could handle it. When an IPC arrives the IPC implementation will:
 // 1) call OnMessageReady() with the tag of the pending IPC. If the dispatcher
-//    returns NULL it means that it cannot handle this IPC but if it returns
+//    returns nullptr it means that it cannot handle this IPC but if it returns
 //    non-null, it must be the pointer to a dispatcher that can handle it.
 // 2) When the  IPC finally obtains a valid Dispatcher the IPC
 //    implementation creates a CrossCallParamsEx from the raw IPC buffer.
@@ -181,26 +182,58 @@
   typedef bool (Dispatcher::*Callback0)(IPCInfo* ipc);
   typedef bool (Dispatcher::*Callback1)(IPCInfo* ipc, void* p1);
   typedef bool (Dispatcher::*Callback2)(IPCInfo* ipc, void* p1, void* p2);
-  typedef bool (Dispatcher::*Callback3)(IPCInfo* ipc, void* p1, void* p2,
+  typedef bool (Dispatcher::*Callback3)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
                                         void* p3);
-  typedef bool (Dispatcher::*Callback4)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4);
-  typedef bool (Dispatcher::*Callback5)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4, void* p5);
-  typedef bool (Dispatcher::*Callback6)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4, void* p5, void* p6);
-  typedef bool (Dispatcher::*Callback7)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4, void* p5, void* p6,
+  typedef bool (Dispatcher::*Callback4)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4);
+  typedef bool (Dispatcher::*Callback5)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4,
+                                        void* p5);
+  typedef bool (Dispatcher::*Callback6)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4,
+                                        void* p5,
+                                        void* p6);
+  typedef bool (Dispatcher::*Callback7)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4,
+                                        void* p5,
+                                        void* p6,
                                         void* p7);
-  typedef bool (Dispatcher::*Callback8)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4, void* p5, void* p6,
-                                        void* p7, void* p8);
-  typedef bool (Dispatcher::*Callback9)(IPCInfo* ipc, void* p1, void* p2,
-                                        void* p3, void* p4, void* p5, void* p6,
-                                        void* p7, void* p8, void* p9);
+  typedef bool (Dispatcher::*Callback8)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4,
+                                        void* p5,
+                                        void* p6,
+                                        void* p7,
+                                        void* p8);
+  typedef bool (Dispatcher::*Callback9)(IPCInfo* ipc,
+                                        void* p1,
+                                        void* p2,
+                                        void* p3,
+                                        void* p4,
+                                        void* p5,
+                                        void* p6,
+                                        void* p7,
+                                        void* p8,
+                                        void* p9);
 
   // Called from the  IPC implementation when an  IPC message is ready override
-  // on a derived class to handle a set of  IPC messages. Return NULL if your
+  // on a derived class to handle a set of  IPC messages. Return nullptr if your
   // subclass does not handle the message or return the pointer to the subclass
   // that can handle it.
   virtual Dispatcher* OnMessageReady(IPCParams* ipc, CallbackGeneric* callback);